pkg/agent/manager/cache/lru_cache.go

```diff
@@ -283,8 +294,11 @@ func (c *LRUCache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.R
 			delete(c.staleEntries, id)
 		}
 	}
+	agent.IncrementEntriesRemoved(c.metrics, entriesRemoved)
```

Comment:
nit that aliasing the telemetry agents package as something like `agentmetrics` or similar would make the code more clear.
as-is calls like `agent.IncrementEntriesRemoved` make it seem like we doing some core spire-agent logic